cmd/juju/application/deploy.go

```diff
@@ -317,6 +318,11 @@ type DeployCommand struct {
 	// NewAPIRoot stores a function which returns a new API root.
 	NewAPIRoot func() (DeployAPI, error)
 
+	// Trust signifies that the charm should be deployed with access to
+	// trusted credentials. That is, hooks run by the charm can access
+	// cloud credentials and other trusted access credentials.
+	Trust bool
```

Comment:
I'm a little concerned that the flag you pass is:
 juju deploy --trust
but that actually links to
 juju elevate-privileges
and
 juju config elevated-privileges

is "juju deploy --elevate-privileges" or "juju deploy --with-elevated-privileges" is overly verbose?